flower/api/prometheus_metrics.py

```diff
@@ -0,0 +1,77 @@
+import logging
+from typing import List, Tuple, Set
+
+from prometheus_client import Counter as PrometheusCounter, Histogram, Gauge
+from prometheus_client.metrics import MetricWrapperBase
+
+logger = logging.getLogger(__name__)
+
+
+class LabelNames:
+    WORKER = 'worker'
+    TYPE = 'type'
+    TASK = 'task'
+
+
+class PrometheusMetrics(object):
+    events = PrometheusCounter(
+        'flower_events_total', "Number of events", [LabelNames.WORKER, LabelNames.TYPE, LabelNames.TASK]
+    )
+    runtime = Histogram('flower_task_runtime_seconds', "Task runtime", [LabelNames.WORKER, LabelNames.TASK])
+    prefetch_time = Gauge(
+        'flower_task_prefetch_time_seconds',
+        "The time the task spent waiting at the celery worker to be executed.",
+        [LabelNames.WORKER, LabelNames.TASK]
+    )
+    number_of_prefetched_tasks = Gauge(
+        'flower_worker_prefetched_tasks',
+        'Number of tasks of given type prefetched at a worker',
+        [LabelNames.WORKER, LabelNames.TASK]
+    )
+    worker_online = Gauge('flower_worker_online', "Worker online status", [LabelNames.WORKER])
+    worker_number_of_currently_executing_tasks = Gauge(
+        'flower_worker_number_of_currently_executing_tasks',
+        "Number of tasks currently executing at a worker",
+        [LabelNames.WORKER]
+    )
+
+    @property
+    def transient_metrics(self) -> List[MetricWrapperBase]:
```

Comment:
If we think that this should be configurable, maybe we could make a second PR and add it as an additional feature. For now I think it makes total sense just to wipe any existing records from metrics for label sets containing offline workers. Otherwise the graphs are polluted with a reading from like a few days ago...